ComponentKitTestHelpers/CKComponentLifecycleTestController.mm

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "CKComponentLifecycleTestController.h"
+
+#import "CKComponentDataSourceAttachController.h"
+#import "CKComponentLayout.h"
+#import "CKComponentProvider.h"
+#import "CKComponentScopeRoot.h"
+#import "CKComponentSizeRangeProviding.h"
+#import "CKDimension.h"
+#import "CKMutex.h"
+
+@interface CKComponentLifecycleTestController () <CKComponentStateListener>
+@end
+
+@implementation CKComponentLifecycleTestController
+{
+  Class<CKComponentProvider> _componentProvider;
+  id<CKComponentSizeRangeProviding> _sizeRangeProvider;
+  CKComponentScopeRoot *_previousScopeRoot;
+  CKComponentStateUpdateMap _pendingStateUpdates;
+  CKComponentLifecycleTestControllerState _state;
+  UIView *_mountedView;
+  CKComponentDataSourceAttachController *_componentDataSourceAttachController;
```

Comment:
`CKComponentLifecycleTestController` now uses `CKComponentDataSourceAttachController` for attaching and detaching component hierarchies to and from views. Now `CKComponentLifecycleTestController` only needs to concern itself with managing component state, building component hierarchies, and laying them out.